load-gen/src/main/java/io/smartcat/data/loader/api/RateGenerator.java

```diff
@@ -0,0 +1,6 @@
+package io.smartcat.data.loader.api;
+
+public interface RateGenerator {
+
+    long getRate(long prevInstant, long currentInstant);
```

Comment:
The notion of time cannot be escaped. The fact that this interface operates with time values clearly shows it. And having it implicitly doesn't mean that it's not there. It could be achieved either explicitly, using introducing a timer (but the constant frequency is of no importance here) , or implicitly, having a repetitive process/loop whose time cycle isn't too short.
In any extent, regardless of the above consideration, a constant rate generator's implementation should be of the form `getRate(...) { return 1000; }`, right?